$OpenBSD: patch-src_main_c,v 1.19 2012/04/06 10:14:44 dcoppa Exp $

Separate delete and destroy events
(upstream svn revisions r2232 & r2233)

Remove deprecated slide_away code to unbreak with glib-2.32
(upstream svn revision r2234)

--- src/main.c.orig	Thu Mar 22 05:58:16 2012
+++ src/main.c	Fri Apr  6 11:42:28 2012
@@ -594,6 +594,7 @@ static void hup_handler(int signum)
 {
     // printf("handling signal %i\n",signum);
     delete_callback(NULL, NULL, NULL);
+    g_idle_add(set_destroy, NULL);
 }
 #endif
 
@@ -734,7 +735,6 @@ int main(int argc, char *argv[])
     subtitle_fuzziness = 0;
     disable_embeddedfonts = FALSE;
     quit_on_complete = FALSE;
-    slide_away = NULL;
     verbose = 0;
     reallyverbose = 0;
     embedding_disabled = FALSE;
@@ -1066,8 +1066,6 @@ int main(int argc, char *argv[])
     create_window(embed_window);
 
     autopause = FALSE;
-    slide_away = g_mutex_new();
-    mplayer_complete_cond = g_cond_new();
 #ifdef GIO_ENABLED
     idledata->caching = g_mutex_new();
     idledata->caching_complete = g_cond_new();
